packages/service-worker/src/provider.ts

```diff
@@ -48,10 +49,13 @@ export function ngswAppInitializer(
       // Wait for service worker controller changes, and fire an INITIALIZE action when a new SW
       // becomes active. This allows the SW to initialize itself even if there is no application
       // traffic.
-      navigator.serviceWorker.addEventListener('controllerchange', () => {
-        if (navigator.serviceWorker.controller !== null) {
-          navigator.serviceWorker.controller.postMessage({action: 'INITIALIZE'});
-        }
+      const onControllerChange = () =>
+        navigator.serviceWorker.controller?.postMessage({action: 'INITIALIZE'});
+
+      navigator.serviceWorker.addEventListener('controllerchange', onControllerChange);
+
+      appRef.onDestroy(() => {
+        navigator.serviceWorker.removeEventListener('controllerchange', onControllerChange);
```

Comment:
nit: for better minification, we can extract the `navigator.serviceWorker` to a separate const:
```suggestion
      const sw = navigator.serviceWorker;
      const onControllerChange = () =>
        sw.controller?.postMessage({action: 'INITIALIZE'});

      sw.addEventListener('controllerchange', onControllerChange);

      appRef.onDestroy(() => {
        sw.removeEventListener('controllerchange', onControllerChange);
```

Reply:
Updated.